On the current failure:

```
    @pytest.mark.django_db
    def test_make_global_user_assignment(admin_api_client, rando, inventory):
        rd = RoleDefinition.objects.create_from_permissions(
            permissions=['change_inventory', 'view_inventory'],
            name='global-change-inv',
            content_type=None,
        )
        url = reverse('roleuserassignment-list')
        data = dict(role_definition=rd.id, user=rando.id, object_id=None)
        response = admin_api_client.post(url, data=data, format="json")
        assert response.status_code == 201, response.data
>       assert response.data['created_by']
E       KeyError: 'created_by'

test_app/tests/rbac/api/test_rbac_views.py:100: KeyError
```

Oof, it's painful because it's correct. When making a POST, there is an established expectation that you get the serialized object in the same way as when it is shown in the list or detail view. For AWX we used to have a jsonschema that verified _exactly this_. You can't do that when it differs from the create serializer. Removing this data in the POST response is exactly what is asked for here.

--- a/ansible_base/lib/dynamic_config/dynamic_settings.py
+++ b/ansible_base/lib/dynamic_config/dynamic_settings.py
@@ -26,8 +26,7 @@ except NameError:
     REST_FRAMEWORK = {}
 
 
-# Tells the base serializer to auto-create a creation-only serializer
-# with the read-only fields excluded, for OpenAPI / openapi-generator / generated client
+# See docs/lib/views.md
 try:
     ANSIBLE_BASE_AUTO_CREATE_SERIALIZER
 except NameError:

--- a/docs/lib/views.md
+++ b/docs/lib/views.md
@@ -48,3 +48,58 @@ class DefaultAPIView(AnsibleBaseView):
     all my good stuff
     ...
 ```
+
+## Automatic Create-only Serializer
+
+For a specific niche case, using this setting will tell the base view
+to auto-create a creation-only serializer with read-only fields excluded.
+
+```
+ANSIBLE_BASE_AUTO_CREATE_SERIALIZER = True
+```
+
+This affects the OpenAPI spec generated in the following way.
+The POST actions for list views will map to a dynamically generated "ModelCreate" serializer
+which will omit the fields which are not editable.
+
+```
+"RoleDefinitionCreate": {
+  "type": "object",
+  "properties": {
+    "permissions": {
+      "type": "array",
+      "items": {
+        "$ref": "#/components/schemas/PermissionsEnum"
+      }
+    },
+    "content_type": {
+      "nullable": true,
+      "description": "The type of resource this applies to\n\n* ...",
+      "oneOf": [
+        {
+          "$ref": "#/components/schemas/ContentTypeEnum"
+        },
+        {
+          "$ref": "#/components/schemas/NullEnum"
+        }
+      ]
+    },
+    "name": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    }
+  },
+  "required": [
+    "name",
+    "permissions"
+  ]
+}
+```
+
+This is specifically offered for using with the openapi-generator
+automatically generated client.
+The "required" listing only shows the writable fields, whereas
+other methods like GET return these every time, making them "required"
+according to jsonschema standards.

--- a/test_app/tests/rbac/api/test_rbac_views.py
+++ b/test_app/tests/rbac/api/test_rbac_views.py
@@ -74,7 +74,11 @@ def test_make_user_assignment(admin_api_client, inv_rd, rando, inventory):
     data = dict(role_definition=inv_rd.id, user=rando.id, object_id=inventory.id)
     response = admin_api_client.post(url, data=data, format="json")
     assert response.status_code == 201, response.data
-    assert response.data['created_by']
+    assert response.data['user'] == rando.pk
+    assert int(response.data['object_id']) == inventory.pk
+    assert response.data['role_definition'] == inv_rd.pk
+
+    assert rando.has_obj_perm(inventory, 'change')
 
 
 @pytest.mark.django_db
@@ -97,7 +101,11 @@ def test_make_global_user_assignment(admin_api_client, rando, inventory):
     data = dict(role_definition=rd.id, user=rando.id, object_id=None)
     response = admin_api_client.post(url, data=data, format="json")
     assert response.status_code == 201, response.data
-    assert response.data['created_by']
+    assert response.data['user'] == rando.pk
+    assert response.data['object_id'] == None
+    assert response.data['role_definition'] == rd.pk
+
+    assert rando.has_obj_perm(inventory, 'change')
 
 
 @pytest.mark.django_db

--- a/test_app/tests/rbac/api/test_rbac_views.py
+++ b/test_app/tests/rbac/api/test_rbac_views.py
@@ -102,7 +102,7 @@ def test_make_global_user_assignment(admin_api_client, rando, inventory):
     response = admin_api_client.post(url, data=data, format="json")
     assert response.status_code == 201, response.data
     assert response.data['user'] == rando.pk
-    assert response.data['object_id'] == None
+    assert response.data['object_id'] is None
     assert response.data['role_definition'] == rd.pk
 
     assert rando.has_obj_perm(inventory, 'change')

